src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs

```diff
@@ -0,0 +1,1021 @@
+                long fixup = pMemoryBlob - (byte*)originalAddress;
+
+                if (request->EntityChunkCount > 0 && dataChunkIndex < request->EntityChunkCount && dataChunkIndex != -1)
+                {
+                    HTTP_DATA_CHUNK* pDataChunk = (HTTP_DATA_CHUNK*)(fixup + (byte*)&request->pEntityChunks[dataChunkIndex]);
+
+                    fixed (byte* pReadBuffer = buffer)
+                    {
+                        byte* pTo = &pReadBuffer[offset];
+
+                        while (dataChunkIndex < request->EntityChunkCount && dataRead < size)
+                        {
+                            if (dataChunkOffset >= pDataChunk->BufferLength)
+                            {
+                                dataChunkOffset = 0;
+                                dataChunkIndex++;
+                                pDataChunk++;
+                            }
+                            else
+                            {
+                                byte* pFrom = pDataChunk->pBuffer + dataChunkOffset + fixup;
+
+                                uint bytesToRead = pDataChunk->BufferLength - (uint)dataChunkOffset;
+                                if (bytesToRead > (uint)size)
+                                {
+                                    bytesToRead = (uint)size;
+                                }
+                                for (uint i = 0; i < bytesToRead; i++)
+                                {
+                                    *(pTo++) = *(pFrom++);
+                                }
+                                dataRead += bytesToRead;
+                                dataChunkOffset += bytesToRead;
+                            }
+                        }
+                    }
+                }
+                //we're finished.
+                if (dataChunkIndex == request->EntityChunkCount)
+                {
+                    dataChunkIndex = -1;
+                }
+            }
+
+            if (NetEventSource.IsEnabled)
+            {
+                NetEventSource.Exit(null);
+            }
+            return dataRead;
+        }
+
+        internal static HTTP_VERB GetKnownVerb(byte[] memoryBlob, IntPtr originalAddress)
+        {
+            NetEventSource.Enter(null);
+
+            // Return value.
+            HTTP_VERB verb = HTTP_VERB.HttpVerbUnknown;
+            fixed (byte* pMemoryBlob = memoryBlob)
+            {
+                HTTP_REQUEST* request = (HTTP_REQUEST*)pMemoryBlob;
+                if ((int)request->Verb > (int)HTTP_VERB.HttpVerbUnparsed && (int)request->Verb < (int)HTTP_VERB.HttpVerbMaximum)
+                {
+                    verb = request->Verb;
+                }
+            }
+
+            NetEventSource.Exit(null);
+            return verb;
+        }
+
+        internal static IPEndPoint GetRemoteEndPoint(byte[] memoryBlob, IntPtr originalAddress)
+        {
+            throw new NotImplementedException();
```

Comment:
Ideally, just fix the code - but I couldn't find a way to do this with the current public IPEndPoint.  PNSE might be appropriate if this ends up being too hard.